src/target/riscv/riscv-013.c

```diff
@@ -5367,6 +5368,14 @@ static enum riscv_halt_reason riscv013_halt_reason(struct target *target)
 		return RISCV_HALT_INTERRUPT;
 	case CSR_DCSR_CAUSE_GROUP:
 		return RISCV_HALT_GROUP;
+	case CSR_DCSR_CAUSE_OTHER:
+		if (get_field(dcsr, CSR_DCSR_EXTCAUSE) == 0) {
+			LOG_TARGET_WARNING(target, "halted because of hart in a critical error state.");
```

Comment:
```suggestion
			LOG_TARGET_INFO(target, "halted because of hart in a critical error state.");
```
I believe `INFO` is more appropriate here, since this is completely normal from the debugger's point of view.

Reply:
Addressed

Reply:
I am sorry but would like to disagree.

Even though this is not an error of OpenOCD or of the debug transport (JTAG adapter or similar), it is situation that is clearly exceptional and requires user attention. As such, it should be prominently displayed to the user. So a warning is in my opinion warranted here.

Reply:
@JanMatCodasip, I will insist)
1. IMHO a similar example will be an assertion failure. GDB does not warn about an assertion failure -- it just notifies you.
2. To be honest, this seems like an issue with the spec, but: what if the `Smdbltrp` extension is not implemented? The `cause` may still be `other` but this is not an exceptional situation.